### ISSUE TITLE
Server controlled skins

### ISSUE THREAD

**Issue by mattmess1221** [MineLittlePony/HDSkins#10]
There should be a packet sent on join to set player skins. This should reduce skin server load and allow the server to control who has what skin, just like vanilla.

A couple security concerns:
- Client should only accept skins from trusted servers
- Texture hosts should be on a whitelist defined by the server, approved by the player.
- Hosts not on the whitelist will not be downloaded.

Skulls
- Skin data can be stored in NBT, just like vanilla.
- Allows skulls to use the same skin as when the player died, just like vanilla.
- Singleplayer will follow the whitelist config to prevent malicious map makers

Progress can be tracked here. https://github.com/MineLittlePony/HDSkins/tree/server-skins

**Comment by Sollace** [comment]
Will you still be able to use custom skins on vanilla servers though?

**Comment by killjoy1221** [comment]
Yes. If the server doesn't provide any skin data, the old method will be
used.

**Comment by Sollace** [comment]
@killjoy1221 Ok, boomer. We'll see where this goes.

Remember to base your branch off 1.15-pre1 if you haven't already.

**Comment by killjoy1221** [comment]
I'll do that once you merge into master.

**Comment by Sollace** [comment]
Happy rebasing then.

**Comment by killjoy1221** [comment]
Basic functionality is working. Just need to tackle security issues and the hostname whitelist.
